src/components/Sticky/StickyTarget.tsx

```diff
@@ -0,0 +1,113 @@
+     * The position of `StickyTarget` is then calculated relative to that element.
+     */
+    getConnectedElement?: (ref: React.MutableRefObject<HTMLDivElement | null>) => Element | false;
+}
+
+/**
+ * Element wraps the content that need to be displayed sticky.
+ * The content then offset relative to its nearest scrolling ancestor and containing block (nearest block-level ancestor).
+ */
+export const StickyTarget = ({
+    className,
+    to = "top",
+    local = false,
+    background = "transparent",
+    offset,
+    style,
+    getConnectedElement,
+    ...otherDivProps
+}: StickyTargetProps) => {
+    const stickyTargetRef = React.useRef<HTMLDivElement | null>(null);
+
+    let offsetStyle = {};
+    if (typeof offset !== "undefined") {
+        offsetStyle = { ...style, "--eccgui-sticky-target-localoffset": offset } as CSSProperties;
+    }
+
+    let connectedOffset = 0;
+    React.useEffect(() => {
+        /**
+         * If the target should be sticky to a defined element then:
+         * * check for the element and its scroll parent
+         * * listen to scroll events and use the elements position as offset
+         */
+        if (getConnectedElement && stickyTargetRef) {
+            const stickyConnection = getConnectedElement(stickyTargetRef);
+            if (stickyConnection) {
+                const scrollParent = utils.getScrollParent(stickyConnection);
+                const scrollParentFallback = !scrollParent ? document.documentElement : false;
+                if (scrollParent || scrollParentFallback) {
+                    const updateTargetOffset = () => {
+                        const scrollParentPosition = (
+                            (scrollParent || scrollParentFallback) as HTMLElement
+                        ).getBoundingClientRect();
+                        const stickyConnectionPosition = stickyConnection.getBoundingClientRect();
+                        if (to === "top") {
+                            connectedOffset =
+                                stickyConnectionPosition.top -
+                                Math.max(0, scrollParentPosition.top) +
+                                stickyConnectionPosition.height;
+                        }
+                        if (to === "bottom") {
+                            connectedOffset =
+                                Math.max(scrollParentPosition.height, scrollParentPosition.bottom) -
+                                stickyConnectionPosition.bottom +
+                                stickyConnectionPosition.height;
+                        }
+                        stickyTargetRef.current?.style.setProperty(
+                            "--eccgui-sticky-target-applicationoffset",
+                            `${connectedOffset}px`
+                        );
+                    };
+                    updateTargetOffset();
+                    (scrollParent || window).addEventListener("scroll", (_event) => {
```

Comment:
Don't we need to removeEventListener in case the react effect dependencies change? Otherwise, we may end up with a case where we have a ton of events attached to a ScrollParent or window - they need to be cleared.

Reply:
You're correct, I forgot that.